build.rs

```diff
@@ -421,42 +394,44 @@ fn get_rustc_link_lib(config: &InterpreterConfig) -> Result<String, String> {
 /// 4. `python{major version}.{minor version}`
 ///
 /// If none of the above works, an error is returned
-fn find_interpreter_and_get_config() -> Result<(InterpreterConfig, HashMap<String, String>), String>
-{
+fn find_interpreter_and_get_config() -> Result<(InterpreterConfig, HashMap<String, String>)> {
     if let Some(sys_executable) = env::var_os("PYTHON_SYS_EXECUTABLE") {
         let interpreter_path = sys_executable
             .to_str()
-            .expect("Unable to get PYTHON_SYS_EXECUTABLE value");
+            .ok_or("Unable to get PYTHON_SYS_EXECUTABLE value")?;
         let interpreter_config = get_config_from_interpreter(interpreter_path)?;
 
-        return Ok((
-            interpreter_config,
-            fix_config_map(get_config_vars(interpreter_path)?),
-        ));
+        return Ok((interpreter_config, get_config_vars(interpreter_path)?));
     };
 
+    let python_interpreter = ["python", "python3"]
+        .iter()
+        .find(|bin| {
+            if let Ok(out) = Command::new(bin).arg("--version").output() {
+                // begin with `Python 3.X.X :: additional info`
+                out.stdout.starts_with(b"Python 3") || out.stderr.starts_with(b"Python 3")
+            } else {
+                false
+            }
+        })
+        .ok_or("Python 3.x interpreter not found")?;
+
     // check default python
-    let interpreter_config = get_config_from_interpreter(&PYTHON_INTERPRETER)?;
+    let interpreter_config = get_config_from_interpreter(&python_interpreter)?;
     if interpreter_config.version.major == 3 {
-        return Ok((
-            interpreter_config,
-            fix_config_map(get_config_vars(&PYTHON_INTERPRETER)?),
-        ));
+        return Ok((interpreter_config, get_config_vars(&python_interpreter)?));
```

Comment:
Removed ` fix_config_map` here since I changed to call it in `get_config_vars` for both targets.